topo/node/cptx/cptx.go

```diff
@@ -361,24 +354,47 @@ func defaults(pb *tpb.Node) *tpb.Node {
 				Inside:   50051,
 				NodePort: node.GetNextPort(),
 			},
-		},
-		Labels: map[string]string{
+		}
+	}
+	if pb.Labels == nil {
+		pb.Labels = map[string]string{
 			"type":   tpb.Node_JUNIPER_CEVO.String(),
 			"vendor": tpb.Vendor_JUNIPER.String(),
-		},
-		Config: &tpb.Config{
-			Image: "cptx:latest",
-			Command: []string{
-				"/entrypoint.sh",
-			},
-			Env: map[string]string{
-				"CPTX": "1",
-			},
-			EntryCommand: fmt.Sprintf("kubectl exec -it %s -- cli -c", pb.Name),
-			ConfigPath:   pb.Config.ConfigPath,
-			ConfigFile:   pb.Config.ConfigFile,
-		},
+		}
+	} else {
+		if pb.Labels["type"] == "" {
+			pb.Labels["type"] = tpb.Node_JUNIPER_CEVO.String()
+		}
+		if pb.Labels["vendor"] == "" {
+			pb.Labels["vendor"] = tpb.Vendor_JUNIPER.String()
+		}
+	}
+	if pb.Config == nil {
+		pb.Config = &tpb.Config{}
+	}
+	if len(pb.Config.GetCommand()) == 0 {
+		pb.Config.Command = []string{
+			"/entrypoint.sh",
+		}
+	}
+	if pb.Config.Image == "" {
+		pb.Config.Image = "cptx:latest"
+	}
+	if pb.Config.Env == nil {
+		pb.Config.Env = map[string]string{
+			"CPTX": "1",
+		}
+	}
```

Comment:
Should this follow the same pattern? Not sure whats best for the env map:

if pb.Config.Env == nil {
pb.Config.Env = map[string]string{}
}
if _, ok := pb.Config.Env["CPTX"]; !ok {
  pb.Config.Env["CPTX"] = "1"
}